src/jit/lower.cpp

```diff
@@ -3737,9 +3746,15 @@ void Lowering::LowerBlock(BasicBlock* block)
     m_block = block;
     m_blockRange = LIR::AsRange(block);
 
-    for (GenTree* node = m_blockRange.FirstNonPhiNode(), *end = m_blockRange.End(); node != end; node = node->gtNext)
+    for (GenTree* node = m_blockRange.FirstNonPhiNode(), *end = m_blockRange.End(), *next; node != end; node = next)
     {
-        LowerNode(node);
+        node = LowerNode(node);
+
```

Comment:
It seems like "decomposed/lowered" node doesn't have a clear definition, whereas "next node to process" does.